content/app/ui/popups/goto-file-popup.js

```diff
@@ -93,10 +117,44 @@ define("content/app/ui/popups/goto-file-popup", function(require, exports, modul
 		 * Trigger submit event.
 		 */
 		_triggerSubmit : function () {
-			var event = new Event(Event.SUBMIT, this._input.value);
+
+			var inputValue = this._input.value;
+			var choosenEntry = null;
+			var entries = this._entries;
+
+			for (var i = 0; i < entries.length; i++) {
+				if (inputValue === entries[i].toString()) {
+					choosenEntry = entries[i];
+					break;
+				}
+			}
```

Comment:
You can make separate function or property with getter that will return currently selected element.